airflow/models/baseoperator.py

```diff
@@ -179,18 +180,49 @@ class derived from this one results in the creation of a task object,
     :param pool_slots: the number of pool slots this task should use (>= 1)
         Values less than 1 are not allowed.
     :type pool_slots: int
-    :param sla: time by which the job is expected to succeed. Note that
-        this represents the ``timedelta`` after the period is closed. For
-        example if you set an SLA of 1 hour, the scheduler would send an email
-        soon after 1:00AM on the ``2016-01-02`` if the ``2016-01-01`` instance
-        has not succeeded yet.
-        The scheduler pays special attention for jobs with an SLA and
-        sends alert
-        emails for sla misses. SLA misses are also recorded in the database
-        for future reference. All tasks that share the same SLA time
-        get bundled in a single email, sent soon after that time. SLA
-        notification are sent once and only once for each task instance.
+    :param sla: Deprecated in favor of ``expected_finish``.
     :type sla: datetime.timedelta
+    :param expected_duration: the maximum duration the task is allowed to take,
+        provided as a ``timedelta``. This ``timedelta`` is relative to when the
+        task actually starts running, not to the execution date. It includes
```

Comment:
Nobody expects duration to start counting from the execution_date, wouldn't mention it to above confusion.

```suggestion
        task actually starts running. It includes
```